FinSearchUnified/Tests/BusinessLogic/Models/FindologicArticleModelTest.php

```diff
@@ -247,4 +247,111 @@ public function articlesWithAttributeProvider()
             ]
         ];
     }
+
+    /**
+     * @dataProvider articleSEOUrlProvider
+     *
+     * @param array $articleConfiguration
+     * @param string $expectedUrl
+     *
+     * @throws ReflectionException
+     */
+    public function testArticleWithSEOUrl(array $articleConfiguration, $expectedUrl)
+    {
+        $baseCategory = new Category();
+        $baseCategory->setId(5);
+
+        $articleFromConfiguration = $this->createTestProduct($articleConfiguration);
+
+        $shop = Manager::getResource('Shop')->getRepository()->find(1);
```

Comment:
Are these two lines actually necessary? You would think that the main shop is already registered.

Reply:
Apparently the conversion does not work without re-registering the shop. If you remove these two lines, it returns the default URL with the `detail/index/sArticle/1` type of format. It seems that there are some other services needed to registered again for the rewrite URL to work.